File: c/dsa/stack.c
#include <stdio.h>
#define MAX 5

int A[MAX], n;

void push(int item) {
    if (n == MAX) printf("overflow!\n");
    else A[n++] = item;
}

void pop() {
    if (n == 0) printf("underflow!\n");
    else n--;
}

void display() {
    for (int i = 0; i < n; i++) printf("%d  ", A[i]);
}

int main() {
    int choice, item;

    printf("1: push; 2: pop; *: exit\n");

    while (1) {
        printf("\n > ");
        scanf("%d", &choice);
        switch (choice) {
        case 1:
            scanf("%d", &item);
            push(item);
            break;
        case 2: pop(); break;
        default: return 0; break;
        }
        display();
    }
}

File: java/Stack.java
class Stack {
    int n = 0; int[] array = new int[5];

    void display() {
        for (int i = 0; i < n; i++)
            System.out.printf("%d ", array[i]);
    }

    void push(int item) {
        if (n == 5) {
            System.out.println("overflow!");
            return;
        }
        array[n++] = item;
    }

    int pop() {
        if (n == 0) {
            System.out.println("overflow!");
            return 0;
        }
        return array[n--];
    }

    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(2);
        stack.push(3);
        stack.pop();
        stack.push(4);
        stack.display();
    }
}
